### ISSUE TITLE
BatchNormalization not consistent with CUNN/NN ?

### ISSUE THREAD

**Issue by misko** [soumith/cudnn.torch#244]
https://gist.github.com/misko/3e61594103c3f47ea1dd8c77405d7301

When I set the network to evaluate mode I am getting significantly different backpropagation values through BatchNormalization on CUDNN vs CUNN/NN in evaluate() mode. Any ideas? maybe a mixup of STD/VAR somewhere?

For larger networks the CUDNN seems to consistently report very small numbers on backward in evaluate() mode, this isn't correct right?

**Comment by szagoruyko** [comment]
right, cudnn doesn't support backward in evaluate mode, we should either add an assert or fall back to nn backward

**Comment by misko** [comment]
Ah! That sucks... is it just subbing in the correct mean/vars, or its internally updating things? I tried hacking that just now into the torch wrapper, but still getting different values :(

**Comment by szagoruyko** [comment]
it's internally updating things. just use nn version, it is not much less efficient

**Comment by misko** [comment]
Thanks for the quick responses! Will do!

**Comment by Amir-Arsalan** [comment]
@soumith  @szagoruyko So there is no way to fix the issue? I mean having a cuDNN network in evaluate mode and do backward pass?

**Comment by szagoruyko** [comment]
@Amir-Arsalan no, cudnn does not support it.
